src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -30,9 +32,30 @@
 (def ServicesStatus
   {schema/Str ServiceStatus})
 
+(def SemVerVersion
+  (schema/pred semver/valid-format? "semver"))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn ^:always-validate
+  get-service-version :- SemVerVersion
+  [group-id artifact-id]
+  (let [version (versioneer/get-version group-id artifact-id)]
+    (when-not version
+      (throw (IllegalStateException.
```

Comment:
I don't think this ever gets triggered, because versioneer returns empty string if it can't find the version.